core/src/main/java/hudson/init/impl/InstallUncaughtExceptionHandler.java

```diff
@@ -37,5 +40,33 @@ public void reportException(Throwable e, ServletContext context, HttpServletRequ
                 }
             }
         });
+        try {
+            Thread.setDefaultUncaughtExceptionHandler(new DefaultUncaughtExceptionHandler());
+            DefaultUncaughtExceptionHandler.LOGGER.log(Level.INFO, "Succesfully installed a global UncaughtExceptionHandler."); 
+        }
+        catch (SecurityException ex) {
```

Comment:
Tomcat applies a minimal set of restrictions IIRC (one of which blocks us capturing the Logs in support-core) 

I would not be surprised if tomcat in addition to its thread monitoring has a rule to watch out for this class loader leak